### PR TITLE
tlmi-auth: init at 1.0.1

### PR DESCRIPTION
**Pull request by snpschaaf**
## Description of changes

> Utility for creating signature strings needed for thinklmi certificate based authentication  

See [`tlmi-auth`](https://github.com/lenovo/tlmi-auth)

Please verify if it is placed well within the nixpkgs folder structure.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tl/tlmi-auth/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    cmake
+    openssl
+  ];
+
+  meta = with lib; {
```
Comment (cideM) [inline pkgs/by-name/tl/tlmi-auth/package.nix:33]: I'm not familiar with the tool but if it has a main executable then adding `mainProgram` would be nice, see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

Comment (snpschaaf) [reply pkgs/by-name/tl/tlmi-auth/package.nix:33]: Thanks for advice. The build results includes only one binary.  The `mainProgram` meta-attribute is defined now.

File: pkgs/by-name/tl/tlmi-auth/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    repo = name;
+    rev = "v${version}";
+    hash = "sha256-/juXQrb3MsQ6FxmrAa7E1f0vIMu1397tZ1pzLfr56M4=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    cmake
```
Comment (nikstur) [inline pkgs/by-name/tl/tlmi-auth/package.nix:29]: Why do we need cmake here? It doesn't look like tlmi-auth uses it.

Comment (snpschaaf) [reply pkgs/by-name/tl/tlmi-auth/package.nix]: Nice catch! it's not used at all. `cmake` is now removed.

**Review by nikstur — APPROVED** [review]

File: pkgs/by-name/tl/tlmi-auth/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    repo = name;
+    rev = "v${version}";
+    hash = "sha256-/juXQrb3MsQ6FxmrAa7E1f0vIMu1397tZ1pzLfr56M4=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    openssl
```
Comment (tfc) [inline pkgs/by-name/tl/tlmi-auth/package.nix:28]: This should be a `buildInputs` to maximize the portability of this package in terms of cross-compilation/static-linking compability etc.

**Comment by tfc** [comment]
Btw. for the record: This is just one new package, but a PR where one employee of the same company merges stuff for his colleague literally on the same day does not seem too healthy for the project longterm.

**Pull request merged**